aea/runtime.py

```diff
@@ -87,7 +87,31 @@ def __init__(
         self._storage: Optional[Storage] = self._get_storage(agent)
 
         self._loop_mode = loop_mode or self.DEFAULT_RUN_LOOP
-        self.main_loop: BaseAgentLoop = self._get_main_loop_instance(self._loop_mode)
+        self._agent_loop: BaseAgentLoop = self._get_agent_loop_instance(self._loop_mode)
+
+    def _log_runtime_state(self, state) -> None:
+        """Log a runtime state changed."""
+        self.logger.debug(f"[{self._agent.name}]: Runtime state changed to {state}.")
+
+    def _get_multiplexer_instance(
+        self, multiplexer_options: Dict, threaded: bool = False
+    ) -> AsyncMultiplexer:
+        """Create multiplexer instance."""
+        loop: Optional[AbstractEventLoop] = None
+        if not threaded:
+            loop = self.loop
+        return AsyncMultiplexer(
+            loop=loop,
+            threaded=threaded,
+            agent_name=self._agent.name,
+            connections=multiplexer_options["connections"],
+            exception_policy=multiplexer_options.get(
+                "connection_exception_policy", ExceptionPolicyEnum.propagate
+            ),
+            default_routing=multiplexer_options.get("default_routing"),
+            default_connection=multiplexer_options.get("default_connection"),
+            protocols=multiplexer_options.get("protocols", []),
```

Comment:
Previously the protocols where missing in the async case. I have no idea why it worked anyway. This is concerning @solarw